src/notifier.ts

```diff
@@ -44,31 +74,47 @@ class Notifier {
    * @method getId
    * @returns the id of this adapter.
    */
-  getId() {
+  getId(): string {
     return this.id;
   }
 
-  getPackageName() {
+  getPackageName(): string {
     return this.packageName;
   }
 
-  getOutlet(id) {
+  getOutlet(id: string): Outlet {
     return this.outlets[id];
   }
 
-  getOutlets() {
+  getOutlets(): Record<string, Outlet> {
```

Comment:
nitpick: I do not have strong opinions but I have seen `: Notifier['outlets']` form being used in the wild, so when you change type of `Notifier#outlets` it will be automatically reflected in getter. That can be both good and a bad thing :-) Good is easier refactoring, bad is exposed API is not set in stone. Personally, I prefer stable API over refactoring tricks.

Reply:
> Good is easier refactoring, bad is exposed API is not set in stone

I would expect the getter to be a stable API from now on.

What would be your solution here?

Reply:
Stable API, so having `getOutlets(): Record<string, Outlet> { /* ... */ }` and `private outlets: Record<string, Outlet> = {};`. Alternatively we could have dedicated

```typescript
type OutletsCollection = Record<string, Outlet>; // didn't thought about name lot ;-)
getOutlets(): OutletsCollection { /* ... */ }
private outlets: OutletsCollection;
```

but that is an overkill, IMO.

Reply:
> Stable API, so having getOutlets(): Record<string, Outlet> { /* ... */ } and private outlets: Record<string, Outlet> = {};

I'm not sure I'm following you.
The variable, as well as the getter, are already of type `Record<string, Outlet>`.

Reply:
And that is what I would use. So all good :-)

Reply:
Perfect